solr/core/src/java/org/apache/solr/cloud/api/collections/CreateCollectionCmd.java

```diff
@@ -116,6 +121,25 @@ public void call(ClusterState clusterState, ZkNodeProps message, @SuppressWarnin
       throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "collection alias already exists: " + collectionName);
     }
 
+    String withCollection = message.getStr(WITH_COLLECTION);
+    String withCollectionShard = null;
+    if (withCollection != null) {
+      String realWithCollection = aliases.resolveSimpleAlias(withCollection);
```

Comment:
Shouldn't this only be done when message says `followAliases` is `true`?

Reply:
Or is the collection created with the `withCollection` property that is assumed to be an actual collection?
Wouldn't this somehow defeats the purpose of an alias? What prevents having the collection `withCollection` property follow aliases? Follow always? Or not always? The current proposal follows always given it doesn't check `followAliases`. In which case maybe resolving the alias each time replicas are added to the collection makes sense? Let the placement plugin do the failure, it's checking the `withCollection` collection is appropriate anyway.

Reply:
Frankly no idea :) that's the way it was done in 8x, there are probably edge cases either way. I'll add a note for now.

Reply:
We should not follow aliases at all and specify that the `withCollection` must be an existing real collection name.